### DISCUSSION TITLE
Need to implement custom middleware to verify user can access that api or not

### DISCUSSION THREAD

**Discussion by nishantbhavasar** [supabase/supabase#19406] [Questions] [answered]
hello,

i have created a application in next.js and i don't have back-end i am managing all of my data directly from supabase client and really i am appreciate that supabase is a best tool ever i have use 

but i am stuck in one place which is very important to build to protect my api's

i have user table which contain a column called active status true or false

now i have to build a middleware like function which we have in every back-end to verify request is user making request is active or not if user inActive need to return access denied from that api 

but in supabase we don't have any feature so i can create a midddleware and set that so it will called before every supabase api call so i can protect my apis 

so please provide me solution for this 
or yes i am not able to update my apis because i have created like 500+ apis or if i am going to update them it will become nightmare for me

if you required any other details please ask me and help me to resolved this

**Comment by GaryAustin1** [comment, marked as answer]
You should be able to just use RLS to check your user status in that table.  You would need to add the RLS to all your tables that you want selected.
https://supabase.com/docs/guides/auth/row-level-security

**Reply by nishantbhavasar** [reply]
@GaryAustin1  thank you very much for explaining me
